### ISSUE TITLE
staticcheck: add lint forbidding casting floats to unsigned integers

### ISSUE THREAD

**Issue by tamird** [dominikh/go-tools#101]
Code and rationale: https://github.com/cockroachdb/cockroach/blob/b8d1c9c8a206541ae590cac7c651f44de2623cde/pkg/cmd/metacheck/main.go#L58:L107

**Comment by dominikh** [comment]
How will we avoid false positives in cases where the user knows that the float is positive?

**Comment by tamird** [comment]
Yeah, we can't. I'm not familiar enough with the convention in this repo regarding certainty - that's also why I opened an issue and not a PR. Feel free to close if the false positive concern makes this infeasible.

**Comment by dominikh** [comment]
I'll leave it open for now, but odds are we won't be able to do it. It *may* go into the dubious category if it has some, but not too many false positives. For all other categories, the rule is "if we flag it, it should be a true positive, or absolutely bizarre code".
